BraveShared/Analytics/DAU.swift

```diff
@@ -38,14 +38,19 @@ public class DAU {
             return false
         }
         
+        // Sending ping immediately
+        sendPingToServerInternal()
+        
+        // Setting up timer to try to send ping after certain amount of time.
+        // This helps in offline mode situations.
         if launchTimer != nil { return false }
         launchTimer =
             Timer.scheduledTimer(
-                timeInterval: activeUserDuration,
+                timeInterval: pingRefreshDuration,
                 target: self,
                 selector: #selector(sendPingToServerInternal),
                 userInfo: nil,
-                repeats: false)
+                repeats: true)
```

Comment:
We should kill this at some point. I think maybe a more flexible approach would be to remove the timer entirely. If the DAU call fails, then we do an `asyncAfter` scheduled for `20.minutes`?

Reply:
Android does it every 20 minutes though, not just once after 20mins. If app is not used for some the timer will be killed anyway

Reply:
It would just be recursive.

- attempt ping
- ping fails, schedule new ping in 20 minutes
- 20 minutes pass
- new ping attempt
- ping fails, schedule new ping in 20 minutes

This way if it is _successful_ it will just automatically stop ping attempts. Or is it supposed to attempt every 20 minutes forever, even if successful? I suppose this would catch situations where users use the app at 11:30pm and are still using it at 12:10am.

Reply:
Android leaves it, so we will too.